src/components/SignUpWidget.js

```diff
@@ -109,8 +106,20 @@ const SignUpWidget = ({
                 setShowTermsStep(true);
                 setShowVerificationStep(false);
               }
+            };
+
+            const signOut = async () => {
+              await firebase.auth().signOut();
+            };
+
+            if (authResult.additionalUserInfo.isNewUser || !signedInUser.emailVerified) {
+              signedInUser.sendEmailVerification();
+              setShowVerificationStep(true);
+              setShowTermsStep(false);
+              signOut().then(() => false);
+            } else if (signedInUser.emailVerified) {
+              successResponse().then(() => false);
```

Comment:
Since the promise is now not being returned and is executed "out of band" of this callback, returning `false` here has no effect.

We solved this in Soul Medicine by:

1. [Always returning `false` in the `signInSuccessWithAuthResult` callback](https://github.com/chaynHQ/soulmedicine/blob/56538d0c24474b8532caaa9f1837f34678115703/app/webpack/components/authentication/sign_in.vue#L205-L207), and
1. [Using the `onAuthStateChanged` Firebase Auth handler to respond to when a user is authed (either through sign up / log in or existing auth credentials](https://github.com/chaynHQ/soulmedicine/blob/56538d0c24474b8532caaa9f1837f34678115703/app/webpack/components/authentication/sign_in.vue#L237)

… I realise that in our case here we use the `authResult` to check if this is a new user, so I'm not sure the best way of refactoring this.